### PR TITLE
Design/dead cleanup

### PR DESCRIPTION
**Pull request by HaicoDorenbos**
Cleanup of some dead code in the Roboteam_utils repository.

### CODE REVIEW

**Review by rolfvdhulst — CHANGES_REQUESTED** [review]
Please don´t delete the Tube class. It´s an important shape for ball placement and will be necessary in the future.

EDIT: Jesse just told me he needed it already but was not aware of it's existence.

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,8 +46,6 @@ add_library(roboteam_utils STATIC
         include/roboteam_utils/pid.h
         src/Triangle.cpp
         include/roboteam_utils/Triangle.h
-        src/Tube.cpp
-        include/roboteam_utils/Tube.h
```
Comment (timovdk) [inline CMakeLists.txt]: Tube is used now

File: CMakeLists.txt [hunk 2]
```diff
@@ -23,14 +23,10 @@ add_library(roboteam_utils STATIC
         include/roboteam_utils/Position.h
         src/Mathematics.cpp
         include/roboteam_utils/Mathematics.h
-        #        src/Cone.cpp
-        #        include/roboteam_utils/Cone.h
```
Comment (timovdk) [inline CMakeLists.txt]: Cone is nice to keep. Tube was unused for the whole year, but as of yesterday it is used. I propose to keep this shape as well for the same reason.

**Review by timovdk — CHANGES_REQUESTED** [review]
Please re-add Tube since Tube is used again as of yesterday

**Comment by rolfvdhulst** [comment]
I would disagree with using cone personally because it is very legacy (I believe from 2016/2017), so you will want to rewrite the code to work properly at some point.

**Comment by HaicoDorenbos** [comment]
I did not know that my PR have such positive influence. By removing code I remind others of possible code clones.

**Comment by HaicoDorenbos** [comment]
Tube has been re-added.

**Comment by rolfvdhulst** [comment]
I feel like this is still missing the point. If the code is tested and works fine, I don't think we should remove it. It's the utils repository after all, and this way people can use it to prevent code clones as you mentioned yourself haico. Especially code that was recently added such as statistics /Triangle and which is tested should really not be removed IMO. Old code that has been unused and untested can go, though.

**Comment by rolfvdhulst** [comment]
In my opinion: if a class/function is tested and works correctly, don't remove it.

**Comment by HaicoDorenbos** [comment]
Don't add new functionality if it is not used directly. Currently the code base is too big which causes often problems when compiling/running the code such as that you cannot run a web browser in parallel to CLion. If a class is used in the future then look back at this branch and re-add it if you need it

**Comment by rolfvdhulst** [comment]
Sure, but all of these classes *should* be used in our code at some point. Also 'usage' can really vary and does not mean they hav to be directly used during our code. I needed the statistics functions a lot when testing roboteam_world earlier this year on branches, where I removed usage later again after the code/models were verified to work correctly for example.

**Comment by rolfvdhulst** [comment]
I disagree with the premise our code base is too big though. It's actually not even close to being big enough to cover all functionality needed to e.g. win the robocup. If running CLion is a problem for you, then I would not see that as a codebase issue but as an IDE issue.

**Comment by HaicoDorenbos** [comment]
> Sure, but all of these classes _should_ be used in our code at some point. Also 'usage' can really vary and does not mean they hav to be directly used during our code. I needed the statistics functions a lot when testing roboteam_world earlier this year on branches, where I removed usage later again after the code/models were verified to work correctly for example.

Theoretically it should be used. But you know what happens in practise, future teams forget that this code exists and the code remains dead...

**Comment by rolfvdhulst** [comment]
And, by continuously asking people to readd the code later, you tend to create a lot more code clones as not everyone is aware of the existence of certain classes. Especially utility like this, should just be available for use IMO. The compile time impact of single files is actually quite minimal (and actually none in the case of header only classes).

**Comment by rolfvdhulst** [comment]
Is it a problem if there is dead but functionally correct code in our system? I don't think so. If it's not correct, then it's a problem and can go.

**Comment by HaicoDorenbos** [comment]
Also dead code causes more development time. Dead code should be properly documented, designed, tested, analysed for bugs, performance optimized, ... 

I am not willing to do that for code that is not used at all

**Comment by HaicoDorenbos** [comment]
But I can make an agreement with you. I can store all the dead code on a separate branch and if someone needs it then he can use it (however then he should also improve the documentation, integrate it in the current design, re-evaluate the test cases, analyse the code for bugs, optimize the code in performance, etc.)

**Comment by rolfvdhulst** [comment]
I would still disagree with that.  Utilities should just be ready for use. If you put them on a seperate branch you might as well remove them as no one ever looks back to old branches unless they are very important (e.g. we played a match on them).

**Comment by HaicoDorenbos** [comment]
So you want me to document/redesign/analyze/optimize/test/etc. all this dead code? I can put a reference in the Wiki to this deadcode branch...

**Comment by HaicoDorenbos** [comment]
We decided to add a dead code comment tag at places where there is dead code

**Comment by HaicoDorenbos** [comment]
All deadcode has been marked. Moreover it is removed from CMakeList to ensure that it is really dead code and to speed up compilation.

**Review by timovdk — APPROVED** [review]

**Review by rolfvdhulst — APPROVED** [review]

**Pull request merged**